### PR TITLE
Civic identity resolver

### PR DESCRIPTION
**Pull request by kevinhcolgan**
Implement a Civic Identity Resolver package using the @civic/profile library to retrieve Civic profile information

### CODE REVIEW

**Review by zhelezkov — CHANGES_REQUESTED** [review]
looks super solid! minor q and gtg :)

File: packages/identity-civic/src/identity-civic.spec.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    const identity = await resolver.resolve(owner);
+
+    expect(identity).toStrictEqual({
+      type: 'CIVIC',
+      name: 'my.nfts',
+      publicKey: owner,
+      additionals: {
+        avatarUrl: 'https://arweave.net/C3oH-InR6fqeqN62TQOI5LnQqNXDmboRK75xFIIT3fU',
+        link: `https://www.civic.me/${owner.toBase58()}`,
+        headline: 'this is my headline'
```
[the remark this answers is not in the document]
Comment (kevinhcolgan) [reply packages/identity-civic/src/identity-civic.spec.ts:20]: the 'headline' is an additional profile information that can be set in civic.me:
image
I added it here for completeness

File: packages/identity-civic/src/identity-civic.spec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  });
+
+  it('should perform a reverse lookup from a .sol address', async () => {
+    const resolver = new CivicIdentityResolver(connection);
+    const owner = new PublicKey('BriW4tTAiAm541uB2Fua3dUNoGayRa8Wt7pZUshUbrPB');
+    const identity = await resolver.resolveReverse('solana.sol');
+
+    expect(identity).toStrictEqual({
+      type: 'CIVIC',
+      name: '',
```
[the remark this answers is not in the document]
Comment (kevinhcolgan) [reply packages/identity-civic/src/identity-civic.spec.ts]: good point, I've pushed a change to use the 'rawDomainName' as the default query name in the case of a reverse lookup

[the remark this answers is not in the document]
Comment (kevinhcolgan) [reply packages/identity-civic/src/identity-civic.spec.ts:32]: on second thoughts, we discussed internally and we think that only things that are present on the Civic profile should be returned. If the dApp wants to use the lookup query as the default then that can be implemented further up the stack

[the remark this answers is not in the document]
Comment (zhelezkov) [reply packages/identity-civic/src/identity-civic.spec.ts:32]: ok, i see your point, sounds good

**Review by zhelezkov — APPROVED** [review]
lgtm

**Pull request merged**